src/pages/VideoPlayerPage.jsx

```diff
@@ -11,10 +11,12 @@ import { toast } from "react-toastify";
 
 const VideoPlayerPage = () => {
   const { playerId } = useParams();
-  const { videosData, videoDispatch } = useVideo();
+  const { videosData, videoState, videoDispatch } = useVideo();
+  const searchVideo = (arr, video) => arr.includes(video);
   const details = findInVideos(videosData, "channelId", playerId);
+
   useEffect(
-    () => videoDispatch({ type: "ADD_TO_HISTORY", payload: details }),
+    () => !searchVideo(videoState["history"],details) && videoDispatch({ type: "ADD_TO_HISTORY", payload: details }),
```

Comment:
can destruct for ease of use. like `{ videoState : { history } }`

Reply:
ok